### DISCUSSION TITLE
what does "<auto>" do in the file actions?

### DISCUSSION THREAD

**Discussion by armyers** [ibhagwan/fzf-lua#1441] [Q&A] [answered]
Hi,

I'm pretty new to the advanced features of neovim and to lua.

I see this block of code (and several like it) in the file `lua/fzf-lua/actions.lua`:

```
-- file actions
M.file_edit = function(selected, opts)
  local vimcmd = "<auto>"
  M.vimcmd_entry(vimcmd, selected, opts)
end
```

I've googled and searched the neovim help pages but cannot find any explanation of what `"<auto>"` does.

Please help! :)

Thank you!

**Comment by ibhagwan** [comment]
> I've googled and searched the neovim help pages but cannot find any explanation of what `"<auto>"` does.

That’s because it’s meant to be used only internally inside fzf-lua :)

Until not long ago there were different code paths for buffer entries (buffers, lines, tabs, etc) and file entries (files, git_file, lsp_xxx, grep_xxx, etc), I combined the actions code for both and now `file_edit = buf_edit`. This simplified setting up the defaults in `actions.files` which now apply to both buffers and files (no need to define `actions.buffers`).

In short, `<auto>` just tells the combined action code to use `:b <bufnr>` and `:e <filepath>` respectively:
https://github.com/ibhagwan/fzf-lua/blob/985e1002b5dc70f8196ea0605bbfffc8c8407e31/lua/fzf-lua/actions.lua#L129-L130

The internal code deduplication resulting from said change:
https://github.com/ibhagwan/fzf-lua/blob/985e1002b5dc70f8196ea0605bbfffc8c8407e31/lua/fzf-lua/actions.lua#L369-L377

**Comment by armyers** [comment, marked as answer]
OK, that's cool. thank you!

so, I know the default behavior for file_edit when multiple files are selected in the picker is to add them to the quckfix list. I'd like to add an option to just open the files in separate buffers (not windows or tabs) so I can skip the step of working through the quickfix list. Does that sound like a reasonable use-case (workflow) and would that be relatively easy? :)

Thanks again!

**Reply by armyers** [reply]
btw, nice plugin! I love it! :)

**Reply by ibhagwan** [reply]
> OK, that's cool. thank you!
> 
> so, I know the default behavior for file_edit when multiple files are selected in the picker is to add them to the quckfix list. I'd like to add an option to just open the files in separate buffers (not windows or tabs) so I can skip the step of working through the quickfix list. Does that sound like a reasonable use-case (workflow) and would that be relatively easy? :)
> 
> Thanks again!

The default action is `file_edit_or_qf`, what you want is also possible by using `file_edit` instead - either change the default in `actions.files` or add a new key bind mapped to `file_edit`.

**Reply by armyers** [reply]
ah, yes, that makes sense now. I just wasn't quite sure because I didn't know the function of "<auto>". and yes, I misspoke about the default behavior of `file_edit`; I was thinking about file_edit_or_qf, I guess.

thank you, again, for your great plugin and quick assistance.

**Reply by ibhagwan** [reply]
Ty @armyers!

Link to the README’s default actions with comments about the qf vs no qf:
https://github.com/ibhagwan/fzf-lua/blob/985e1002b5dc70f8196ea0605bbfffc8c8407e31/README.md?plain=1#L658-L676

**Reply by armyers** [reply]
yes, I found that earlier and was just reviewing. I'll probably add a `C-b` mapping to call `file_edit` to open the selected files in buffers, like you suggested.  Very cool.

Once again, thank you!

**Reply by armyers** [reply]
and fyi, that will remove a `ton` of friction from my workflow! I've been feeling a bit frustrated by not having a way to directly open the files in buffers that you (and telescope) have made so easy to find. Good stuff!

**Comment by armyers** [comment]
sorry, one more quick follow up. Here's my code:
```
return {
  {
    "ibhagwan/fzf-lua",
    enabled = true,
    -- optional for icon support
    dependencies = { "nvim-tree/nvim-web-devicons" },
    config = function()
      local actions = require("fzf-lua.actions")
      -- calling `setup` is optional for customization
      require("fzf-lua").setup({
        actions = {
          files = {
            true, -- inherit from defaults
            ["enter"] = actions.file_edit,
          },
        },
      })
    end,
  },
}
```

I stuck with using `enter` cuz I couldn't find a good alternative `ctrl-` combo. :/

I love it!

**Reply by ibhagwan** [reply]
looks great @armyers!
